src/os_factory.cc

```diff
@@ -23,9 +23,6 @@ OsLayer *OsLayerFactory(const std::map<std::string, std::string> &options) {
   os = new OsLayer();
 
   // Check for memory allocation failure.
-  if (!os) {
-    logprintf(0, "Process Error: Can't allocate memory\n");
-    return 0;
-  }
+  if (!os) return 0;
```

Comment:
Why not to add an error here? At least this requires some error logging.

Reply:
This is handled in the outer class. No need to throw two errors.